gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/BloodPressureMapper.java

```diff
@@ -0,0 +1,237 @@
+import static uk.nhs.adaptors.pss.translator.util.ParticipantReferenceUtil.getParticipantReference;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.StringUtils;
+import org.hl7.fhir.dstu3.model.CodeableConcept;
+import org.hl7.fhir.dstu3.model.DateTimeType;
+import org.hl7.fhir.dstu3.model.Encounter;
+import org.hl7.fhir.dstu3.model.Identifier;
+import org.hl7.fhir.dstu3.model.Observation;
+import org.hl7.fhir.dstu3.model.Observation.ObservationComponentComponent;
+import org.hl7.fhir.dstu3.model.Observation.ObservationStatus;
+import org.hl7.fhir.dstu3.model.Patient;
+import org.hl7.fhir.dstu3.model.Period;
+import org.hl7.fhir.dstu3.model.Reference;
+import org.hl7.fhir.dstu3.model.UriType;
+import org.hl7.v3.CD;
+import org.hl7.v3.II;
+import org.hl7.v3.RCMRMT030101UK04Annotation;
+import org.hl7.v3.RCMRMT030101UK04Component02;
+import org.hl7.v3.RCMRMT030101UK04Component3;
+import org.hl7.v3.RCMRMT030101UK04Component4;
+import org.hl7.v3.RCMRMT030101UK04CompoundStatement;
+import org.hl7.v3.RCMRMT030101UK04EhrComposition;
+import org.hl7.v3.RCMRMT030101UK04EhrExtract;
+import org.hl7.v3.RCMRMT030101UK04NarrativeStatement;
+import org.hl7.v3.RCMRMT030101UK04ObservationStatement;
+import org.hl7.v3.RCMRMT030101UK04PertinentInformation02;
+import org.springframework.stereotype.Service;
+
+import lombok.AllArgsConstructor;
+import uk.nhs.adaptors.pss.translator.util.BloodPressureValidatorUtil;
+
+@Service
+@AllArgsConstructor
+public class BloodPressureMapper {
+    private static final String META_PROFILE = "https://fhir.nhs.uk/STU3/StructureDefinition/CareConnect-GPC-Observation-1";
+    private static final String IDENTIFIER_SYSTEM = "https://PSSAdaptor/";
+    private static final String BATTERY_VALUE = "BATTERY";
+    private static final String SYSTOLIC_NOTE = "Systolic Note: ";
+    private static final String DIASTOLIC_NOTE = "Diastolic Note: ";
+    private static final String BP_NOTE = "BP Note: ";
+
+    private CodeableConceptMapper codeableConceptMapper;
+
+    public List<Observation> mapBloodPressure(RCMRMT030101UK04EhrExtract ehrExtract, Patient patient, List<Encounter> encounters) {
+        /**
+         * TODO: Known future implementations to this mapper
+         * - performer: fallback to a default 'Unknown User' Practitioner if none are present in performer (NIAD-2026)
+         * - identifier: concatenate source practice org id to identifier URL (NIAD-2021)
+         */
+
+        var compositionsList = getCompositionsContainingCompoundStatement(ehrExtract);
+
+        return compositionsList.stream()
+            .flatMap(ehrComposition -> ehrComposition.getComponent().stream())
+            .map(RCMRMT030101UK04Component4::getCompoundStatement)
+            .filter(Objects::nonNull)
+            .filter(compoundStatement -> BATTERY_VALUE.equals(compoundStatement.getClassCode().get(0))
```

Comment:
not sure if this reads better or worse to you but you could change your flatmap into more actions for possibly better readability
```return compositionsList.stream()
            .map(RCMRMT030101UK04EhrComposition::getComponent)
            .flatMap(List::stream)
            .map(RCMRMT030101UK04Component4::getCompoundStatement)
            .filter(Objects::nonNull)
```